src/fakes/ray/rpc/raylet/raylet_client.h

```diff
@@ -48,30 +56,166 @@ class FakeRayletClient : public RayletClientInterface {
   void ReleaseUnusedActorWorkers(
       const std::vector<WorkerID> &workers_in_use,
       const rpc::ClientCallback<rpc::ReleaseUnusedActorWorkersReply> &callback) override {
+    num_release_unused_workers += 1;
+    release_callbacks.push_back(callback);
   }
 
   void CancelWorkerLease(
       const TaskID &task_id,
-      const rpc::ClientCallback<rpc::CancelWorkerLeaseReply> &callback) override {}
+      const rpc::ClientCallback<rpc::CancelWorkerLeaseReply> &callback) override {
+    num_leases_canceled += 1;
+    cancel_callbacks.push_back(callback);
+  }
+
+  bool GrantWorkerLease() {
+    return GrantWorkerLease("", 0, WorkerID::FromRandom(), node_id_, NodeID::Nil());
+  }
+
+  bool GrantWorkerLease(const std::string &address,
+                        int port,
+                        const WorkerID &worker_id,
+                        const NodeID &node_id,
+                        const NodeID &retry_at_node_id,
+                        Status status = Status::OK(),
+                        bool rejected = false) {
+    rpc::RequestWorkerLeaseReply reply;
+    if (!retry_at_node_id.IsNil()) {
+      reply.mutable_retry_at_raylet_address()->set_ip_address(address);
+      reply.mutable_retry_at_raylet_address()->set_port(port);
+      reply.mutable_retry_at_raylet_address()->set_node_id(retry_at_node_id.Binary());
+    } else {
+      reply.mutable_worker_address()->set_ip_address(address);
+      reply.mutable_worker_address()->set_port(port);
```

Comment:
```suggestion
    rpc::RequestWorkerLeaseReply reply;
    reply.mutable_retry_at_raylet_address()->set_ip_address(address);
    reply.mutable_retry_at_raylet_address()->set_port(port);
    if (!retry_at_node_id.IsNil()) {
      reply.mutable_retry_at_raylet_address()->set_node_id(retry_at_node_id.Binary());
    } else {
```

Not needed since we're just shuffling lines around for this PR, but it stuck out at me.

Reply:
big true. if CI doesn't pass this time and I need to push again I'll fix